ibis/pandas/execution.py

```diff
@@ -573,42 +583,74 @@ def execute_not_bool(op, data, **kwargs):
 }
 
 
+def compute_join_column(column_expr, **kwargs):
+    column_op = column_expr.op()
+    if isinstance(column_op, ops.TableColumn):
+        name, new_column = column_op.name, None
+    else:
+        name = util.guid()
+        new_column = execute(column_expr, **kwargs)
+
+    root_table, = column_op.root_tables()
+    return name, new_column, root_table
+
+
```

Comment:
shouldn't these be functions rather than computed at import time?

Reply:
This serves only to be a unique suffix that we can depend on to use for selecting out duplicate column names in a `Selection` operation. As a counterexample, if there were a unique suffix for every join operation then we wouldn't be able to select out any overlapping columns.